### PR TITLE
Update `SyncHelper` - Remove 3 phpstan issues

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- [x] Get rid of 3 PHPStan issues.

### CODE REVIEW

File: phpstan/baseline.neon [hunk 1]
```diff
@@ -5610,4 +5595,3 @@ parameters:
 			message: "#^Only booleans are allowed in a ternary operator condition, array\\<string\\> given\\.$#"
 			count: 1
 			path: ../tests/Composer/Test/Util/TlsHelperTest.php
-
```
Comment (stof) [inline phpstan/baseline.neon:5613]: this should not be removed as phpstan generates the baseline with this empty line in it.

Comment (drupol) [reply phpstan/baseline.neon:5613]: Oops! I will fix this as soon as Github works again on my laptop :)

Comment (drupol) [reply phpstan/baseline.neon:5613]: I just double checked and there's an empty line already.

image

Comment (stof) [reply phpstan/baseline.neon:5613]: This is not the empty line on which I commented

Comment (drupol) [reply phpstan/baseline.neon:5613]: I just noticed, fix is incoming.

Comment (drupol) [reply phpstan/baseline.neon]: Fixed.

**Review by stof — APPROVED** [review]

**Comment by Seldaek** [comment]
Thanks

**Comment by drupol** [comment]
@Seldaek I think we could improve this helper further.

What do you think about replacing the `DownloaderInterface` with `DownloadManager` in [the function signature](https://github.com/drupol/composer/blob/7528c86e58884f5676d5851d52cd94c67372d609/src/Composer/Util/SyncHelper.php#L32).

If we do that, we would benefit of the very useful logic in `DownloadManager::download` where the source of the package is automatically detected.

The only drawback here is that instead of relying on an interface, we rely on a concrete implementation... but nothing prevent us to create an interface for the `DownloadManager`.

WDYT ?

**Comment by Seldaek** [comment]
I don't need an interface there, but this is also a breaking change. If we can accept both for BC then I guess why not.

**Comment by stof** [comment]
The BC layer is easy: instead of replacing the interface with DownloadManager, change the argument to accept `DownloadManager|DownloaderInterface`.

**Comment by drupol** [comment]
Yes, but this is not supported in PHP 7.2, this is why I did it in a different way: https://github.com/composer/composer/pull/11496

**Comment by stof** [comment]
Well, you can put the union type in phpdoc (with a check inside the method if you want to keep a TypeError on invalid type) instead of having 2 methods in the API.

**Pull request merged**